### PR TITLE
Validate Grafana Alertmanager configuration

### PR DESCRIPTION
**Pull request by santihernandezc**
We currently accept any string as a valid grafana Alertmanager configuration. This PR uses the `PostableApiAlertingConfig` struct from [grafana/alerting](https://github.com/grafana/alerting/blob/adfb25200d3e7b67633de704211db5191b65bf65/definition/alertmanager.go#L196-L201) and validates the received config before accepting it.

### CODE REVIEW

File: pkg/alertmanager/api_grafana.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/pkg/errors"
 	"github.com/prometheus/alertmanager/cluster/clusterpb"
 
+	"github.com/grafana/alerting/definition"
```
Comment (stevesg) [inline pkg/alertmanager/api_grafana.go:17]:
```suggestion
	"github.com/prometheus/alertmanager/cluster/clusterpb"
	"github.com/grafana/alerting/definition"

```

You need to move this up to the next section to make the linter happy.

File: pkg/alertmanager/api_grafana.go [hunk 2]
```diff
@@ -56,7 +57,7 @@ func (gc *UserGrafanaConfig) Validate() error {
 		return errors.New("created must be non-zero")
 	}
 
-	return nil
+	return gc.GrafanaAlertmanagerConfig.AlertmanagerConfig.Validate()
 }
```
Comment (stevesg) [inline pkg/alertmanager/api_grafana.go:61]: [ultra-nit-xxl] I preferred the pattern of `if`s followed by `return nil` at the end, but that's stylistic and I won't block on it.

**Comment by stevesg** [comment]
Overall LGTM, please take a look at some minor nits, fix the vendoring conflicts and linter.

**Review by stevesg — APPROVED** [review]
Thanks for addressing the nits.

**Review by pracucci — APPROVED** [review]

**Pull request merged**